sensory_jamming.py

```diff
@@ -0,0 +1,342 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Apr 04 14:31:31 2016
+
+@author: Claire
+"""
+
+from __future__ import division
+import random as rd
+import math  as m
+import numpy as np
+import matplotlib.pyplot as plt
+
+# import pylab
+
+class Launcher:
+    # Class implementation to initiate simulatory environment.
+
+    def __init__(self, popsize, boxsize, d_t, simduration):
+        self.popsize = int(popsize)
+            # integer. Size of the bat population / Number of agents 
+        self.d_t = float(d_t * m.pow(10,-3))
+            # float. Time resolution, i.e. how much time (in ms) 
+            # is expressed in 1 simulation time step.
+        self.simduration = int(simduration)
+            # integer. Total number of time steps to be run in the simulation. 
+        self.realtime = 0 
+            # time of the simulation, in seconds. Initially set at 0.
+        self.timeclock = np.empty(self.simduration, dtype = float)
+            # empty array for floats, dimensions 1*simduration. Will 
+            # contain all times in seconds for each simulation time step.
+        self.timeclock[0] = float(self.realtime) 
+            # record the first time, i.e. 0.
+        self.realduration = float(self.simduration * self.d_t)
+            # duration of the simulation in seconds.
+        self.all_ID = np.empty(self.popsize, dtype = int)
+            # empty array for integers, dimensions 1*popsize. Will 
+            # contain the identification numbers of all the agents to be run.
+        self.all_initpos = np.empty([self.popsize,2], dtype = float)
+            # empty array of intial positions for the whole population.
+        self.callsources = {}
+            # empty dictionary. Will contain the sources of each calls, 
+            # emitted by every agent throughout the simulation.
+        self.boxsize = boxsize 
+            # list of 2 integers. Area (rectangle) within which the agents can 
+            # move, defined by the lengths of edges in meters. 
+            # Has to be of the form [x,y], which is asserted below:
+        
+        assert isinstance(self.boxsize, list) and len(self.boxsize) == 2, "'boxsize' must be a list of 2 elements." 
+            # returns error if boxsize is not of the right format, i.e. [x,y]
+        
```

Comment:
@thejasvibr You mean for Identification only, or for every function in the code? Because there is already a detailed description of the functions and classes in the README file. I thought it make the code lighter, and if you need the detailed description you still have access to it.